src/Microsoft.ML.Data/Evaluators/RankerEvaluator.cs

```diff
@@ -573,8 +573,8 @@ internal Result(IExceptionContext ectx, IRow overallResult)
             {
                 VBuffer<double> Fetch(string name) => Fetch<VBuffer<double>>(ectx, overallResult, name);
 
-                Dcg = Fetch(RankerEvaluator.Dcg).Values;
-                Ndcg = Fetch(RankerEvaluator.Ndcg).Values;
+                Dcg = Fetch(RankerEvaluator.Dcg).GetValues().ToArray();
+                Ndcg = Fetch(RankerEvaluator.Ndcg).GetValues().ToArray();
             }
```

Comment:
Hmmm. I wonder if this was another bug.

Reply:
Yes, this was another bug that I fixed (caught by a test).

---
In reply to: [232433774](https://github.com/dotnet/machinelearning/pull/1580#discussion_r232433774) [](ancestors = 232433774)